crates/nu-command/src/filesystem/rm.rs

```diff
@@ -468,8 +468,32 @@ fn rm(
             }
         })
         .filter(|x| !matches!(x.get_type(), Type::Nothing))
+        .fold((Vec::new(), Vec::new()), |(mut out, mut err), v| {
+            match v {
+                Value::Error { error } => err.push(*error),
+                val => out.push(val),
+            }
+
+            (out, err)
+        });
+
+    if let Err(e) = out
+        .into_iter()
         .into_pipeline_data(ctrlc)
-        .print_not_formatted(engine_state, false, true)?;
+        .print_not_formatted(engine_state, false, true)
+    {
+        err.push(e);
+    }
 
-    Ok(PipelineData::empty())
+    match err.len() {
```

Comment:
the code you've added looks very similar. Do you think it's possible to extract it to separate code and reuse it here?